ocaml/runtime/major_gc.c

```diff
@@ -996,10 +1002,10 @@ void caml_major_collection_slice (intnat howmuch)
   }
   if (p < dp) p = dp;
   if (p < caml_extra_heap_resources) p = caml_extra_heap_resources;
-  p += p_backlog;
-  p_backlog = 0.0;
+  p += backlog_words / Caml_state->stat_heap_wsz;
```

Comment:
This is integer division, right?
```suggestion
  p += (double) backlog_words / Caml_state->stat_heap_wsz;
```